src/game/inject.c

```diff
@@ -0,0 +1,721 @@
+#include "game/inject.h"
+
+#include "config.h"
+#include "filesystem.h"
+#include "game/gamebuf.h"
+#include "game/output.h"
+#include "game/packer.h"
+#include "global/const.h"
+#include "global/vars.h"
+#include "log.h"
+#include "memory.h"
+#include "util.h"
+
+#include <stddef.h>
+
+#define BIN_VERSION 1
+
+typedef enum INJECTION_TYPE {
+    INJ_GENERAL = 0,
+    INJ_BRAID = 1,
+} INJECTION_TYPE;
```

Comment:
Yes, `INJ_GENERAL` would probably be targeted at default injections, if there are any like this in the future (like any data "bugs" we'd always want to fix). For others, there is flexibility where we can create further general types, or have super-specific ones. So there might be a general `INJ_TEXTURE_FIX` type in the future, linked to `g_Config.fix_og_texture_issues` for example, and for this there could be a `caves_texture_fixes.bin`, `vilcabamba_texture_fixes.bin` etc injection files. Or there might be a very specific texture fix in a level that we may not want to bundle with the general one, so that would have its own type and config option (texture fixes maybe aren't the best example but hopefully you know what I mean).
Happy to talk about a different approach to linking the injections to config options though if we feel this is a bit clunky.

Reply:
I'm OK with this the way it is at the moment, I think it's quite clean and better than anything I was able to think of ATM.